core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -1057,6 +1059,28 @@ protected Object visitFunctionCall(FunctionCall node, Object context)
             return functionInvoker.invoke(resolvedFunction, connectorSession, argumentValues);
         }
 
+        @Override
+        protected Object visitTrim(Trim node, Object context)
+        {
+            ImmutableList.Builder<Type> argumentTypes = ImmutableList.builder();
+            ImmutableList.Builder<Expression> argumentValues = ImmutableList.builder();
+
+            argumentTypes.add(type(node.getTrimSource()));
+            argumentValues.add(node.getTrimSource());
+
+            node.getTrimCharacter().ifPresent(trimCharacter -> {
+                argumentTypes.add(CodePointsType.CODE_POINTS);
+                argumentValues.add(new Cast(trimCharacter, toSqlType(CodePointsType.CODE_POINTS)));
+            });
+
+            FunctionCall functionCall = FunctionCallBuilder.resolve(session, metadata)
+                    .setName(QualifiedName.of(node.getSpecification().getFunctionName()))
+                    .setArguments(argumentTypes.build(), argumentValues.build())
+                    .build();
+
+            return visitFunctionCall(functionCall, context);
+        }
```

Comment:
It doesn't work, because the function is not resolved.

Anyway, I think we shouldn't try to interpret `Trim`. `ExpressionInterpreter` is generally used in the Planner, where `Trim` should not be present.

In some cases we use `ExpressionInterpreter` to evaluate a constant expression before it ever reaches the Planner (so it could come across a `Trim` node). But in those cases, it fails to evaluate FunctionCalls, so it would be consistent to fail for `Trim`.